keyboards/tssouthpaw/keymaps/default/keymap.c

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2025 TS Design Works LLC
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ #include QMK_KEYBOARD_H
+ #include "rgb_effects/rgb_effects.h"
+ #include "rgb_effects/work_timer.h"
+  
+ // Define layers
+ enum layers {
+     BASE, // Base layer
+     FN,   // Function layer
+ };
+  
+ // Define custom keycodes
+ enum custom_keycodes {
+     KC_MICMUTE = SAFE_RANGE, // Microphone mute key
+     KC_MEDIA,                // Media layer toggle
+     KC_WRKTMR,               // Work timer toggle
+     KC_WRKPAU,               // Work timer pause/resume
+     KC_CPSWRD,               // Caps word toggle
+     KC_30MIN,                // 30-minute timer
+     KC_1HR,                  // 1-hour timer
+     KC_4HR,                  // 4-hour timer
+     KC_8HR,                  // 8-hour timer
+     KC_10HR,                 // 10-hour timer
+ };
+ 
+ // Function to handle rotary encoder updates
+ bool encoder_update_user(uint8_t index, bool clockwise) {
+     if (clockwise) {
+         tap_code(KC_VOLU); // Rotate right: Volume up
+     } else {
+         tap_code(KC_VOLD); // Rotate left: Volume down
+     }
+     return true;
+ }
```

Comment:
This should be replaced with encoder map. 
```c
#ifdef ENCODER_MAP_ENABLE
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
  [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
  [1] = { ENCODER_CCW_CW(_______, _______) },
};
#endif
```
Which also requires `ENCODER_MAP_ENABLE = yes` in a rules.mk, alongside the keymap.c file